src/frontend/src/eth/services/send.services.ts

```diff
@@ -484,20 +469,75 @@ const prepareAndSignApproval = async ({
 	return { success: true, hash };
 };
 
+const resetExistingApprovalToZero = async (
+	params: Omit<SignAndApproveParams, 'amount'>
+): Promise<
+	ResultSuccess & {
+		hash?: string;
+	}
+> =>
+	await prepareAndSignApproval({
+		...params,
+		amount: ZERO
+	});
+
+const checkExistingApproval = async ({
+	token,
+	from,
+	spender,
+	amount,
+	sourceNetwork,
+	...rest
+}: Omit<ApproveParams, 'to' | 'minterInfo' | 'progress'> & {
+	nonce: number;
+	spender: EthAddress;
+}): Promise<{ existingApprovalIsEnough: boolean; approvalNeededReset: boolean }> => {
```

Comment:
The case `true` & `true` is not handled, so instead of two boolean, I was thinking, can we return a `type`?

Something like:

```
result: "existingApprovalsEnough" | "approveNeededReset" | "none"
```

WDYT?

Reply:
Yes, better! Changed!